packages/grid/x-data-grid/src/components/panel/filterPanel/GridFilterForm.tsx

```diff
@@ -102,6 +103,23 @@ const FilterFormValueInput = styled(FormControl, {
   overridesResolver: (_, styles) => styles.filterFormValueInput,
 })({ width: 190 });
 
+interface SelectOptionProps {
+  isNative: boolean;
+  children: React.ReactNode;
+  value: string;
+}
+
+const SelectOption = ({ isNative, children, value, ...props }: SelectOptionProps) =>
+  isNative ? (
+    <option value={value} {...props}>
+      {children}
+    </option>
+  ) : (
+    <MenuItem value={value} {...props}>
+      {children}
+    </MenuItem>
+  );
```

Comment:
Since this code is duplicated, maybe a dedicated file `components/SelectOption.tsx` could be more appropriate